### DISCUSSION TITLE
Will this automatically work with new `flake8` releases?

### DISCUSSION THREAD

**Discussion by hiAndrewQuinn** [john-hen/Flake8-pyproject#4] [General]
Hiya John -- I'm working on a [big forkin' modern Python checklist](https://leafy-taiyaki-59fd85.netlify.app/#flake8-pep8-python-code-linter) and I came across `flake8-pyproject`.

Tremendous work. I'd love to be able to advise my readers that they can just outright remove the `pre-commit` git hook that calls vanilla `flake8` and just call this instead. But what if `flake8` **comes out with a new major version?** Will this TOML wrapper break?

As a speculative question, I figured you'd know better than me. You authored the code after all.

**Comment by john-hen** [comment]
Hi Andrew. I don't think it would break. Obviously I cannot be 100% certain, but I'm definitely banking on that myself and don't expect I have to change any code whatsoever as new Flake8 releases come out, even major ones. (Though there haven't been any so far, it seems, latest release on PyPI right now is from October 2021.)

It wasn't actually much work at all: There is very little code, you can have a look yourself, it's all in [`hook.py`](https://github.com/john-hen/Flake8-pyproject/blob/3f7ddae6e40a4378f4c43e35f5ee1d4fce022174/flake8p/hook.py). We do hook into Flake8 internals, so Flake8 devs could decide to somehow break that if they wanted to. But given their demonstrated reluctance to include this very little thing themselves, for fear of the "maintenance burden" or something, it's exceedingly unlikely they'd so much as touch that part of their code.

**Reply by john-hen** [reply]
So, turns out the unexpected happened and Flake8 5.0.0, released yesterday, did refactor the way they load configuration files, which breaks the hook we're using. For now, I've had to pin Flake8 to an earlier version.

Oh, and of course Flake8 **still** does not support `pyproject.toml`, because that's "too much work" and they're too busy changing all the other code to do the same thing as before.

**Reply by john-hen** [reply]
Fixed in Flake8-pyproject 1.0.0, released today.

**Comment by hiAndrewQuinn** [comment]
Wow. Glad you had the chance to patch it in so quick at least.

---
LI:         https://www.linkedin.com/in/heiAndrewQuinn/
GH:          https://github.com/hiAndrewQuinn

Personal website:    https://andrew-quinn.me/




Aug 1, 2022, 13:37 by ***@***.***:
